crypto/property/property.c

```diff
@@ -582,6 +726,18 @@ int ossl_method_store_fetch(OSSL_METHOD_STORE *store,
     } else {
         ret = 0;
     }
+
+    OSSL_TRACE_BEGIN(QUERY) {
+        char buf[512];
+        int size;
+
+        size = ossl_property_list_to_string(NULL, pq, buf, 512);
+        BIO_printf(trc_out, "method store query with properties %s "
+                   "resolves to provider %s\n",
+                   size == 0 ? "none" : buf,
+                   ossl_provider_name(best_impl->provider));
```

Comment:
How do we know `best_impl` is not NULL here? In other words: my concern is we may be running here even if we found nothing.

I think code here should be moved to else branch at line 723 or we should re-check local variable `ret`. If ret is zero then trace should say no provider matching query could be found.